tensorflow_addons/optimizers/adabelief.py

```diff
@@ -19,9 +19,14 @@
 
 from typing import Union, Callable, Dict
 
+if tf.__version__[:3] > "2.8":
```

Comment:
I'have some doubt to repeat this block in every optimizer file 

@seanpmorgan What do you think? Can we define this as `optimizer_legacy` in the optimizers init.py?

Reply:
I think we should avoid duplicating this code.

Please note that the condition is wrong because it fails for TF 2.10:
```
>>> tf.__version__
'2.10.0-dev20220531'
>>> tf.__version__[:3]
'2.1'
>>> tf.__version__[:3] > "2.8"
False
```